scripts/service_install.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+service_name="gatestext.service"
```

Comment:
These shell scripts will succeed even if there's an error here. Can you please add `set -euxo pipefail` at the beginning of the script please (after the shebang) for all the scripts.

Reply:
Updated but not on the service_enable script, this is killing the bash not to continue due to non zero exit status return by grep if pattern not found. That is expected behavior by grep and already have a check in the script to run next commands or not.

Reply:
That's still problematic because we might unknowingly be eating up other errors in the script (say for example we have a problem in systemctl or something). 

If you add an `else` condition then it would not exit right? Try doing that?

Reply:
I agree but the `set `doesn't know about logical conditions. All it does is instructs bash to immediately exit if any command has a non-zero exit status. The `else ` block doesn't solve the problem. Since this has only systemctl and grep command, I wouldn't want to investigate more at this point to figure out workaround.

Reply:
I'm sorry I disagree :)
Even though it seems inconsequential now, it might turn into a bigger pain later. And I would prefer to have stronger foundation now as compared to later. If we know of a potential bug, I would rather fix it today than wait for it to actually blow up and then fix it then.

Generally something like `grep 'fail' || echo 'did not succeed'` works even for `set -e`. If the `else` condition is not working, maybe rewrite the script to use these operators instead? Something like this maybe? 

```bash
grep --quiet XYZ ||
   (  ... do things ...
   )
```

Reply:
Thanks for pointers, fixed it.